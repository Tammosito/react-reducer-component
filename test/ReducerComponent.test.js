import React from "react";
import ReducerProvider from "../src/index";
import renderer from "react-test-renderer";

const INC_COUNTER = "INC_COUNTER";
const incCounter = () => {
	return {
		type: INC_COUNTER
	};
};

const reducer = (state = {}, action) => {
	if (action.type === INC_COUNTER) {
		return {
			...state,
			count: state.count + 1
		};
	} else return state;
};

const initialState = {
	count: 0
};

test("Render without errors", () => {
	const component = renderer.create(
		<ReducerProvider
			initialState={initialState}
			reducer={reducer}
			render={(reduce, props) => (
				<button onClick={() => reduce(incCounter())}>
					{props.count}
				</button>
			)}
		/>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test("Should reduce state properly", () => {
	const component = renderer.create(
		<ReducerProvider
			initialState={initialState}
			reducer={reducer}
			render={(reduce, props) => (
				<button onClick={() => reduce(incCounter())}>
					{props.count}
				</button>
			)}
		/>
	);

	let tree = component.toJSON();
	// trigger reduce to update state
	tree.children[0].props.onClick();

	tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
