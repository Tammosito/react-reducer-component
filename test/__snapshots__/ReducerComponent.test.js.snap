// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render without errors 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    0
  </button>
</div>
`;

exports[`Should reduce state properly 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    1
  </button>
</div>
`;
